File: 백준/Silver/1475. 방 번호/방 번호.py
import math
number = input().rstrip()
ret = [0] * 10

for n in number:
  ret[int(n) - 1] += 1
tmp = math.ceil((ret[5] + ret[8]) / 2)
ret[5] = ret[8] = tmp
print(max(ret))

File: 백준/Silver/1475. 방 번호/방 번호.java
import java.io.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

        int[] ret = new int[10];
        String num = br.readLine();
        for (char n: num.toCharArray()) {
            int idx = n != '9' ? n - '0' : 6;
            ret[idx] += 1;
        }
        ret[6] = (int)Math.round((double)ret[6] / 2);

        int maxCnt = 0;
        for (int r : ret) {
            maxCnt = Math.max(maxCnt, r);
        }
        bw.write(maxCnt + '0');
        bw.flush();
        br.close(); bw.close();
    }
}